src/arrayMethodSort.js

```diff
@@ -4,8 +4,29 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+    const defaultCompare = (el1, el2) => {
+      const itemOne = `${el1}`;
+      const itemTwo = `${el2}`;
+
+      return itemOne > itemTwo ? +1 : -1;
+    };
+
+    const comparer = arguments.length === 0 ? defaultCompare : compareFunction;
+
+    for (let i = 0; i < this.length; i++) {
+      for (let n = i + 1; n < this.length; n++) {
+        if (comparer(this[i], this[n]) > 0) {
+          const buffer = this[i];
+
+          this[i] = this[n];
+          this[n] = buffer;
+        }
+      }
+    }
```

Comment:
The bubble sort algorithm is correctly implemented, but it is not the most efficient sorting algorithm for larger arrays. Consider implementing a more efficient algorithm, like quicksort or mergesort, for better performance with larger data sets.